proxy-robots-module/src/main/java/orwell/proxy/ProxyRobots.java

```diff
@@ -176,7 +184,7 @@ private void onDefault() {
     public void start() {
         this.connectToServer();
         if (robotsMap.getRobotsArray().isEmpty())
-            this.initializeTanksFromConfig();
+            this.initializeRobotsFromConfig();
         this.connectToRobots();
         //We have to start the communication service before sending Register
         //Otherwise we risk not being ready to read Registered in time
```

Comment:
With a thread I do not think this is enough to be sure it is in the right state.